test/groovy/CheckForLegacyConfigurationTest.groovy

```diff
@@ -126,6 +126,68 @@ class CheckForLegacyConfigurationTest extends BasePiperTest {
                                   "This configuration option was removed. "])
     }
 
+    @Test
+    void testCheckForMissingConfigKeys() {
+        nullScript.commonPipelineEnvironment.configuration = [steps: [someStep: [:]]]
+        Map configChanges = [importantConfigKey: [steps: ['someStep'], customMessage: "test"]]
+
+        List errors = stepRule.step.checkForLegacyConfiguration.checkForMissingConfigKeys(nullScript, configChanges)
+
+        assertEquals(errors, ["Your pipeline configuration does not contain the configuration key importantConfigKey for the step someStep. test"])
+    }
+
+    @Test
+    void testCheckForMissingConfigKeysWithWarning() {
+        String expectedWarning = "[WARNING] Your pipeline configuration does not contain the configuration key importantConfigKey for the step someStep. test"
+
+        nullScript.commonPipelineEnvironment.configuration = [steps: [someStep: [:]]]
+        Map configChanges = [importantConfigKey: [steps: ['someStep'], warnInsteadOfError: true, customMessage: "test"]]
+
+        List errors = stepRule.step.checkForLegacyConfiguration.checkForMissingConfigKeys(nullScript, configChanges)
+        assertEquals(expectedWarning, echoOutput)
+        assertEquals(errors, [])
+    }
+
+    @Test
+    void testCheckForMissingStageConfigKeys() {
+        nullScript.commonPipelineEnvironment.configuration = [stages: [someStage: [:]]]
+        Map configChanges = [importantConfigKey: [stages: ['someStage']]]
+
+        List errors = stepRule.step.checkForLegacyConfiguration.checkForMissingConfigKeys(nullScript, configChanges)
+
+        assertEquals(errors, ["Your pipeline configuration does not contain the configuration key importantConfigKey for the stage someStage. "])
+    }
+
+    @Test
+    void testCheckForMissingGeneralConfigKeys() {
+        nullScript.commonPipelineEnvironment.configuration = [general: [:]]
+        Map configChanges = [importantConfigKey: [general: true]]
+
+        List errors = stepRule.step.checkForLegacyConfiguration.checkForMissingConfigKeys(nullScript, configChanges)
+
+        assertEquals(errors, ["Your pipeline configuration does not contain the configuration key importantConfigKey in the general section. "])
+    }
+
+    @Test
+    void testCheckForPresentGeneralConfigKeys() {
+        nullScript.commonPipelineEnvironment.configuration = [general: [importantConfigKey: 'isPresent']]
+        Map configChanges = [importantConfigKey: [general: true]]
+
+        List errors = stepRule.step.checkForLegacyConfiguration.checkForMissingConfigKeys(nullScript, configChanges)
+
+        assertEquals(errors, [])
+    }
+
+    @Test
+    void testCheckForMissingPostActionConfigKeys() {
```

Comment:
Potentially missing is a test which checks whether it does not fail in case the config is there.

Reply:
I've added one test which checks this for the general section, [here](https://github.com/SAP/jenkins-library/pull/2332/files#diff-e0fc53bf2a3cb8332b37e63c62fd4f35e09bf96c5198c659e5903f9c60fb603bR172). At the time, I didn't see much value to add inverted tests for all supported sections, since the code is the same for each. I can add these tests, though, if you prefer.